src/internal/packager/helm/post-render.go

```diff
@@ -34,16 +37,27 @@ type renderer struct {
 func (h *Helm) newRenderer() (*renderer, error) {
 	message.Debugf("helm.NewRenderer()")
 
-	namespaces := make(map[string]*corev1.Namespace)
-	if h.cluster != nil {
-		namespaces[h.chart.Namespace] = h.cluster.NewZarfManagedNamespace(h.chart.Namespace)
+	rend := &renderer{
+		Helm:           h,
+		connectStrings: types.ConnectStrings{},
+		namespaces:     map[string]*corev1.Namespace{},
+	}
+	if h.cluster == nil {
+		return rend, nil
 	}
 
-	return &renderer{
-		Helm:           h,
-		connectStrings: make(types.ConnectStrings),
-		namespaces:     namespaces,
-	}, nil
+	namespace, err := h.cluster.Clientset.CoreV1().Namespaces().Get(context.TODO(), h.chart.Namespace, metav1.GetOptions{})
```

Comment:
I think we are trying to avoid using `context.TODO()` if possible

```suggestion
	namespace, err := h.cluster.Clientset.CoreV1().Namespaces().Get(context.Background(), h.chart.Namespace, metav1.GetOptions{})
```

Reply:
IMO it fits here, since we want a context to be passed through instead at some point in the future, assuming we intend to add it context to the create and dev flows.

Reply:
I think this is the right place to use it as it is a function missing a context in its signature, but we want to add a context in the future.